core/src/Kokkos_View.hpp

```diff
@@ -92,26 +92,56 @@ constexpr KOKKOS_INLINE_FUNCTION std::size_t count_valid_integers(
          (i6 != KOKKOS_INVALID_INDEX) + (i7 != KOKKOS_INVALID_INDEX);
 }
 
-KOKKOS_INLINE_FUNCTION
-void runtime_check_rank(const size_t rank, const size_t dyn_rank,
-                        const bool is_void_spec, const size_t i0,
-                        const size_t i1, const size_t i2, const size_t i3,
-                        const size_t i4, const size_t i5, const size_t i6,
-                        const size_t i7, const std::string& label) {
+template <typename View>
+KOKKOS_INLINE_FUNCTION void runtime_check_rank(
+    const View&, const size_t rank, const size_t dyn_rank,
+    const bool is_void_spec, const size_t i0, const size_t i1, const size_t i2,
+    const size_t i3, const size_t i4, const size_t i5, const size_t i6,
+    const size_t i7, const std::string& label) {
   (void)(label);
 
   if (is_void_spec) {
     const size_t num_passed_args =
         count_valid_integers(i0, i1, i2, i3, i4, i5, i6, i7);
+    // We either allow to pass as many extents as the dynamic rank is, or
+    // as many extents as the total rank is. In the latter case, the given
+    // extents for the static dimensions must match the
+    // compile-time extents.
+    const bool n_args_is_dyn_rank = num_passed_args == dyn_rank;
+    const bool n_args_is_rank     = num_passed_args == rank;
+
+    if (n_args_is_rank) {
+      size_t new_extents[8] = {i0, i1, i2, i3, i4, i5, i6, i7};
+      for (size_t i = dyn_rank; i < rank; ++i)
+        if (new_extents[i] != View::static_extent(i)) {
+          ranks_matching = false;
+          KOKKOS_IF_ON_HOST(
+              const std::string message =
+                  "The specified run-time dimension for Kokkos::View '" +
```

Comment:
The specified run-time extent for Kokkos::View '`label`' does not match the compile-time extent along the `i`th dimension.